internal/cli/cmd/login/login.go

```diff
@@ -128,24 +124,113 @@ func (o *option) authAndSetToken() error {
 	return nil
 }
 
-func (o *option) readAndSetCredentialInput() error {
-	if len(o.username) == 0 {
-		_, _ = fmt.Fprint(o.writer, "Username: ")
-		user, err := read.FromStdin()
-		if err != nil {
-			return err
+type MyWriteCloser struct {
+	writer io.Writer
+}
+
+func (c MyWriteCloser) Write(p []byte) (n int, err error) {
+	return c.writer.Write(p)
+}
+
+func (c MyWriteCloser) Close() error {
+	return nil
+}
+
+func (o *option) readAndSetLoginInputNative() func() error {
+	return func() error {
+		o.loginKind = loginKindNative
+		if len(o.username) == 0 {
+			prompt := promptui.Prompt{
+				Label:  "Username",
+				Stdout: MyWriteCloser{writer: o.writer},
+			}
+			user, err := prompt.Run()
+			if err != nil {
+				return err
+			}
+			o.username = user
 		}
-		o.username = user
-	}
-	if len(o.password) == 0 {
-		_, _ = fmt.Fprint(o.writer, "Password: ")
-		password, err := readPassword()
-		if err != nil {
-			return err
+		if len(o.password) == 0 {
+			prompt := promptui.Prompt{
+				Label:  "Password",
+				Stdout: MyWriteCloser{writer: o.writer},
+				Mask:   ' ',
```

Comment:
I remember that this was unwanted by @Nexucis: we see the number of character of the password
![image](https://github.com/perses/perses/assets/10258608/4aa62490-e2c4-4a53-ab61-9b5c9c1f8406)

I will see if I can find a lib that really hides it or get back the readpassword function

Reply:
Used huh instead, that should support it soon https://github.com/charmbracelet/huh/issues/129